light-node/src/rpc.rs

```diff
@@ -38,13 +38,18 @@ mod sealed {
     use jsonrpc_derive::rpc;
 
     use tendermint_light_client::supervisor::Handle;
+    use tendermint_light_client::types::LatestStatus;
     use tendermint_light_client::types::LightBlock;
 
     #[rpc]
     pub trait Rpc {
         /// Returns the latest trusted block.
         #[rpc(name = "state")]
         fn state(&self) -> FutureResult<Option<LightBlock>, Error>;
+
+        /// Returns the latest status.
+        #[rpc(name = "status")]
+        fn status(&self) -> FutureResult<Option<LatestStatus>, Error>;
```

Comment:
Result for this endpoint looks like this:
```json
{
  "jsonrpc": "2.0",
  "result": {
    "block_hash": "1BA78A40F4D785E6F30E01E1B706546EAB0B88B6005BE8E175198E74C3E53F40",
    "connected_nodes": [
      "BADFADAD0BEFEEDC0C0ADEADBEEFC0FFEEFACADE",
      "CEFEEDBADFADAD0C0CEEFACADE0ADEADBEEFC0FF"
    ],
    "height": 2,
    "valset_hash": "74F2AC2B6622504D08DD2509E28CE731985CFE4D133C9DB0CB85763EDCA95AA3"
  },
  "id": 1
}
```

Unfortunately, it does not seem to update and always remains the same height?! But at least the height is always on par with the other RPC endpoint :-D  @romac is currently investigating.

Reply:
Just noting that there seems to be a bug with the light store. it doesn't seem to update:
```rust
pub fn trust_block(&mut self, lb: &LightBlock) {
        self.state.light_store.update(lb, Status::Trusted);
        dbg!(lb.height(), self.latest_trusted().unwrap().height());
}
```

Reply:
Maybe changing the `latest_trusted` to something like @liamsi @romac 
``` 
/// in SledStore
fn latest(&self, status: Status) -> Option<LightBlock> {
    self.db(status)
        .iter(&self.db)
        .max_by(|first, second| first.height().cmp(&second.height()))
}
/// in MemoryStore
fn latest(&self, status: Status) -> Option<LightBlock> {
    self.store
        .iter()
        .filter(|(_, e)| e.status == status)
        .max_by_key(|(&height, _)| height)
        .map(|(_, e)| e.light_block.clone())
}
```
This is not good performance for sure but might be worth trying to see if the nature previous usage of iterators was wrong. Since I don;t think the sled one forces ordering on inserts.

Reply:
Thanks!

Reply:
At least that does something:
```
 cargo run -- start                                                                                                                      15:32:09
    Finished dev [unoptimized + debuginfo] target(s) in 0.18s
     Running `[...]/tendermint-rs/target/debug/light-node start`
started RPC server: 127.0.0.1:8888
error: sync failed: no witness left
error: sync failed: no witness left
error: sync failed: no witness left
error: sync failed: no witness left
error: sync failed: no witness left
error: sync failed: no witness left
error: sync failed: no witness left
error: sync failed: no witness left
error: sync failed: no witness left
^C^J                                                     
```

Reply:
Actually, that fixes the bug. I had to clean up the previous dangling light store. @OStevan you're a hero  👍  😄